scraping_class/scraping_class.py

```diff
@@ -185,6 +186,7 @@ def get(self, url, project_name):
         else:
             t = time.time()
             ratelimit()
+            url = url +'?{}'.format('&'.join(['{0}={1}'.format(k, v) for (k, v) in params.items()]))
```

Comment:
We need to encode the query so the browser understands special characters. Try to implement it using `quote` from `urllib.parse`. 

The second think we need to consider is, what if we have an url which already has begun its query like http://random.url?key1=value. The code you have written would produce the url http://random.url?key1=value?key2=value2 (where params equals `{"key": "value2"}`). The url we want to produce is http://random.url?key1=value&key2=value2

Reply:
Maybe factor the url construction into a new function `construct_query` - this would make it easier to write a few tests for it.

Reply:
Okay, I'll give it a shot! Will a '?' followed by characters always indicate that the url already has parameter values? In other word can a url have multiple question marks?

Reply:
One ? is the maximum as it signifies the beginning of a string of key-value options. Theres an in depth explanation on [wikipedia](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Generic_syntax). 

We dont need to cover the entire URL format, we just need to make sure that if a ? is already in the url when it reaches line 189 it doesn't blindly append another ?, but adds the options properly. 

I think the most straightforward way is to do this with [urllib.urlparse](https://docs.python.org/3/library/urllib.parse.html)

Reply:
Also note we just merged some changes to the logging stuff, so you might want to repull before putting in to much work :)

Reply:
I've made a test function to ensure that `construct_query` works. Should I append this test function to test_scraping_class.py file?

Reply:
Jakob have changed a single line in your submission which makes it a bit easier to read. Other than that it looks super.

About the test - yep, you can just add it to `tests/test_scraping_class.py` and add another commit with the updated tests. 

If you then do 
```
pip install pytest
```
and run `pytest` from your terminal in the main folder of the repo, it should run all the tests and give you a short summary of the results (come ask us if this needs elaboration).